### PR TITLE
Update supply control to burn genesis ATN supply

### PR DESCRIPTION
**Pull request by ScottRMalley**
Under normal operation, ATN repaid on a CDP is returned to the SupplyControl contract, where it can be minted on a subsequent `borrow` call. This change updates the SupplyControl contract to send an initial amount `burnableGenesisSupply` to the zero address, rather than keeping it in circulation.

This ensures that over time, all ATN on the network will correspond exactly to debt on a CDP position.

In order to achieve this:
* `SupplyControl` constructor is initialized with a new variable `burnableGenesisSupply`
* `burn` on `SupplyControl` is updated to send to the zero address while `burnableGenesisSupply > 0`
* contract deployment is updated to deploy `SupplyControl` with a `burnableGenesisSupply` value derived from the GenesisAlloc

### CODE REVIEW

File: autonity/solidity/contracts/asm/SupplyControl.sol [hunk 1]
```diff
@@ -90,6 +95,16 @@ contract SupplyControl is ISupplyControl {
...
+            uint256 _toBurn = msg.value;
+            if (msg.value > burnableGenesisSupply) {
+                _toBurn = burnableGenesisSupply;
+                burnableGenesisSupply = 0;
+            } else {
+                burnableGenesisSupply -= msg.value;
+            }
+            payable(address(0)).transfer(_toBurn);
+        }
         emit Burn(msg.value);
```
Comment (Jason-Zhangxin-Chen) [inline autonity/solidity/contracts/asm/SupplyControl.sol:108]: This legacy code looks wired, it does not burn anything at all by emitting a message telling value is burnt. Shall we need to move this `emit Burn(msg.value)` into the added logic block where the burning happens?

**Review by Jason-Zhangxin-Chen — COMMENTED** [review]
LGTM, just leave one comment.

**Review by tbssajal — APPROVED** [review]
LGTM!

File: autonity/solidity/contracts/asm/SupplyControl.sol (hunk 1 above)
Comment (vpiyush) [inline autonity/solidity/contracts/asm/SupplyControl.sol:108]: Shouldn't we emit _toBurn in Burn event instead of msg.value?

**Pull request closed without merging**